### ISSUE TITLE
Post coverage data in QUnit.done callback

### ISSUE THREAD

**Issue by btamas** [JackFallows/qunit-puppeteer-runner#1]
Hi,

first of all, I really like your tool. Thank you for it :)
Although, I face with a problem when I want to send test coverage data to my webserver. I injected a script into a test.html to POST instanbul `__coverage__` object from window, but my callback runs after `qunit-puppeter-runner`'s QUnit.done callback that closes `puppeteer`. Sometimes I got coverage data, sometimes I don't, because `puppeteer` is closed before the POST request is finished.
Do you have an idea how can I be sure my callback runs before `puppeteer` closes? I cannot add my callback earlier than `qunit-puppeteer-runner` adds its callback, because it is added in a QUnit object setter.

**Comment by btamas** [comment]
As I see `QUnit.on('runEnd', function(){})` runs before your callback and I got my coverage data everytime.

**Comment by JackFallows** [comment]
Hi, thanks for checking out the tool. I’m glad you found a workaround for your issue.

I’m not overly familiar with the istanbul tool, but it sounds like you do need Puppeteer to be running to get test the test coverage data (I.e. it’s not enough to just have the results of the test run?).

If it would still be useful, I can certainly look into the possibility of allowing custom QUnit.done callbacks that run before the built in callback that closes Puppeteer?

Thanks

**Comment by btamas** [comment]
What I did is a small middleware that instrument the source code with istanbul. Istanbul will add log methods into the source to measure a line or a scope does run or not. In the end, there will be an object on window called `__coverage__`, that will contain the result. If I would post this JSON to my webserver I would create a report from it.

The mentioned solution is working now, but I have a feeling that it is volatile, because if QUnit behaviour will be modified in the future, maybe it will break. So if you would be that nice to provide a possibility to have a callback before you close Puppeteer, I would be really grateful for it.

**Comment by JackFallows** [comment]
I've just committed a change that should work for you. Essentially, in the options you pass to `initialise`, a new `qunitCallbacks` object can be passed, e.g.
```
qunitCallbacks: {
  done: function () {
    // do something
  }
}
```
I've published the new v1.0.0-rc.5 release to npm. I hope this works as you would expect.

Thanks

**Comment by btamas** [comment]
Thank you very much @JackFallows ;)
